src/walletConnect/v2/saga.ts

```diff
@@ -345,21 +350,22 @@ export function* acceptSession({ session }: AcceptSession) {
   yield call(handlePendingStateOrNavigateBack)
 }
 
-function* getSessionFromClient(session: SignClientTypes.EventArguments['session_proposal']) {
+function* getSessionFromClient(session: Web3WalletTypes.EventArguments['session_proposal']) {
   if (!client) {
     // should not happen
     throw new Error('missing client')
   }
 
-  let sessionValue = client.session.values.find(
-    (value) => value.peer.publicKey === session.params.proposer.publicKey
-  )
-
+  let sessionValue: null | Web3WalletTypes.SessionProposal = null
```

Comment:
i think we should try to assign this instead of declaring as null - the current flow you have is
1) sessionValue = null
2) try to assign session value
3) wait 500ms 
4) repeat from 2 if 2 failed to assign a value

there will always be a 500ms wait in the above loop, even if step 2 passes the first time. while not terribly blocking, it's still an unnecessary delay...

Reply:
Fixed in fade04c04e203e8cf7e4b7b142a760e4626f2a26!